### PR TITLE
MG-316: Add available_data[] to model_overview

### PR DESCRIPTION
**Pull request by beatrizsaldana**
### Problem
The model_overview transform output lacks a clear summary of what data types are available for each model.

### Solution
Added a new `available_data` field to the model_overview transform output that provides a comprehensive list of available data types for each model.

**Key Changes:**
1. **New Utility Function**: Created `get_list_of_available_data()` that analyzes a model's data availability and returns human-readable data type labels
2. **Enhanced Transform Logic**: Updated `transform_model_overview()` to calculate and include the `available_data` field for each model record
3. **Output Schema Update**: Added `available_data` to the final transformed output columns

**Data Mapping:**
- `gene_expression` → "Gene Expression"
- `disease_correlation` → "Disease Correlation"
- `pathology` → "Pathology"
- `biomarkers` → "Biomarkers"

**Example Output:**
```json
{
  "name": "3xTg-AD",
  "model_type": "Familial AD",
  "available_data": ["Gene Expression", "Pathology", "Biomarkers"],
  // ... other fields
}
```

### Testing
**Unit Tests Added:**
- `test_all_data_available()`: Verifies correct mapping when all data types are present
- `test_some_data_missing()`: Ensures proper handling when only some data types are available
- `test_all_data_missing()`: Validates empty list return when no data is available
- `test_empty_model()`: Tests edge case with empty model data
- `test_partial_keys()`: Verifies behavior when only some data keys are present

**Integration Tests Updated:**
- Updated all existing test output files to include the new `available_data` field
- Verified correct `available_data` values across different test scenarios

### CODE REVIEW

File: src/agoradatatools/etl/transform/model_overview.py [hunk 1]
```diff
@@ -47,6 +47,32 @@
...
+
+    # Check if links are present (indicating data is available)
+    if "gene_expression" in row and row["gene_expression"] is not None:
+        available_data.append("Gene Expression")
+    if "disease_correlation" in row and row["disease_correlation"] is not None:
+        available_data.append("Disease Correlation")
+    if "pathology" in row and row["pathology"] is not None:
+        available_data.append("Pathology")
+    if "biomarkers" in row and row["biomarkers"] is not None:
+        available_data.append("Biomarkers")
```
Comment (thomasyu888) [inline src/agoradatatools/etl/transform/model_overview.py]: Nit: Consider something like this

```
fields = {
    "gene_expression": "Gene Expression",
    "disease_correlation": "Disease Correlation",
    "pathology": "Pathology",
    "biomarkers": "Biomarkers"
}

available_data = [label for key, label in fields.items() if row.get(key) is not None]
```

That said, the way you've written the code is technically easier to follow, but just more tedious to add more fields

File: src/agoradatatools/etl/transform/model_overview.py [hunk 2]
```diff
@@ -113,25 +139,33 @@ def transform_model_overview(
...
         ]
 
-        # Build the links
+        # Helper function to check if a value is truthy (including string "TRUE")
+        def is_truthy_for_links(value):
+            if value is True:
+                return True
+            if isinstance(value, str) and value.strip().upper() == "TRUE":
+                return True
+            return False
```
Comment (thomasyu888) [inline src/agoradatatools/etl/transform/model_overview.py]: Nit: Here, the if-conditions you are calculating actually return the boolean value you want.  So for example:

```
    return value is True or (isinstance(value, str) and value.strip().upper() == "TRUE")
```

^ Not sure if that works, but just pseudo-code.

Comment (thomasyu888) [reply src/agoradatatools/etl/transform/model_overview.py]: nested functions are also harder to write unit tests, but i understand why this is a nested helper function

Comment (thomasyu888) [reply src/agoradatatools/etl/transform/model_overview.py]: The other easier thing to do is...

```
return bool(value)
```

That said, I wonder if you need to worry about other value types or if value will always be boolean and/or 'True', "TRUE". "FALSE", "False"

Comment (beatrizsaldana) [reply src/agoradatatools/etl/transform/model_overview.py]: There were spaces in the test files which were causing `a = " "` --> `bool(a)` --> `true`. I did not notice this and was really struggling to figure out why some things came back true when they shouldn't. Fixed it an am implementing this much simpler solution :)

File: src/agoradatatools/etl/transform/model_overview.py (hunk 1 above)
Comment (beatrizsaldana) [reply src/agoradatatools/etl/transform/model_overview.py]: Implemented. Great idea! Thank you!

File: src/agoradatatools/etl/transform/model_overview.py [hunk 3]
```diff
@@ -47,6 +47,32 @@
...
+        List[str]: A list of available data for the model.
+    """
+    fields = {
+        "gene_expression": "Gene Expression",
+        "disease_correlation": "Disease Correlation",
+        "pathology": "Pathology",
+        "biomarkers": "Biomarkers",
+    }
+
+    available_data = [
```
Comment (linglp) [inline src/agoradatatools/etl/transform/model_overview.py:69]: In the docstring, it states: "If the value is a string (link_url), it is considered available." However, in the actual implementation, it appears that any non-None value is treated as available, regardless of whether it's a string. There seems to be a bit of discrepancy between the docstring and the behavior of the code.

File: src/agoradatatools/etl/transform/model_overview.py [hunk 4]
```diff
@@ -113,25 +139,25 @@ def transform_model_overview(
...
         row["gene_expression"] = (
             {"link_url": f"comparison/expression?model={row['name']}"}
-            if row["gene_expression"] is True
+            if bool(row["gene_expression"])
             else None
         )
         row["disease_correlation"] = (
             {"link_url": f"comparison/correlation?model={row['name']}"}
-            if row["disease_correlation"] is True
+            if bool(row["disease_correlation"])
```
Comment (linglp) [inline src/agoradatatools/etl/transform/model_overview.py:150]: I am not sure if this is intended but `bool(row["disease_correlation"])` is not the same as `if row["disease_correlation"] is True`. 

Let's say if we have the following rows in the a dataframe: 
```
df = pd.DataFrame([
    {"name": "modelA", "disease_correlation": [0.8, 0.9]},   
    {"name": "modelB", "disease_correlation": None},         
    {"name": "modelC", "disease_correlation": 0},            
    {"name": "modelD", "disease_correlation": "yes"},
    {"name": "modelE", "disease_correlation": True},
])
```
 If you have `if row["disease_correlation"] is True`, then only the last one would return `{'link_url': 'comparison/correlation?model=modelE'}`
But if you have `bool(row["disease_correlation"])`, the output would be something like: 
```
{'link_url': 'comparison/correlation?model=modelA'}
None
None
{'link_url': 'comparison/correlation?model=modelD'}
{'link_url': 'comparison/correlation?model=modelE'}
```

File: tests/test_assets/model_overview/input/model_overview_model_results_info_good_test_input.csv [hunk 5]
```diff
@@ -1,4 +1,4 @@
 name,gene_expression,disease_correlation,pathology,biomarkers
 3xTg-AD,TRUE,,TRUE,TRUE
 5xFAD (UCI),TRUE,,TRUE,TRUE
-APOE4,TRUE,TRUE,, 
\ No newline at end of file
+APOE4,TRUE,TRUE,,
```
Comment (linglp) [inline tests/test_assets/model_overview/input/model_overview_model_results_info_good_test_input.csv:4]: It seems like gene_expression, disease_correlation, pathology, and biomarkers columns all have boolean or empty data types in the input test data. Do you know if we need to be testing other input types for these columns?

**Review by linglp — CHANGES_REQUESTED** [review]
Hi @beatrizsaldana ! Thanks for your hard work on this! I left a few comments in the code.

One thing I wanted to flag: it looks like there was a change from `if row["biomarkers"] is True` to `if bool(row["disease_correlation"])`. I'm not sure if this switch was intentional. Could you clarify?

Also, I noticed a small discrepancy between the docstring and the implementation of the `get_list_of_available_data` function. The docstring says a value is considered available if it is a string, but the code seems to treat any non-null value as sufficient. It might be worth updating either the logic or the docstring to ensure consistency.

File: tests/test_assets/model_overview/input/model_overview_model_results_info_good_test_input.csv (hunk 5 above)
Comment (JessterB) [reply tests/test_assets/model_overview/input/model_overview_model_results_info_good_test_input.csv:4]: The input data should contain only boolean or empty values.

**Review by jaclynbeck-sage — COMMENTED** [review]
Everything looks good to me, pending Lingling's comments.

File: src/agoradatatools/etl/transform/model_overview.py (hunk 4 above)
Comment (beatrizsaldana) [reply src/agoradatatools/etl/transform/model_overview.py:150]: I am using bool() because the expected values are "TRUE" and "". When I was using `if row["disease_correlation"] is True` the string "TRUE" was not being captured as `True`. @linglp please let me know if this reasoning is illogical and if you have a better implementation in mind. Thank you for opening up this conversation!

Comment (JessterB) [reply src/agoradatatools/etl/transform/model_overview.py:150]: >  the expected values are "TRUE" and ""

The [source file](https://www.synapse.org/Synapse:syn51942280) was created specifically to support this transform, so if there's a preferred way to specify boolean values in csv I'm open to using it.

Comment (linglp) [reply src/agoradatatools/etl/transform/model_overview.py:150]: Got it. Thanks @beatrizsaldana . If the input values are either "True" or "", using `bool()` should be fine. But just to make sure, if your input is `TRUE` and not `True`, this would result in an error: 
```
(synapsePythonClient) lpeng@Mac synapsePythonClient % python -c "print(bool("TRUE"))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    print(bool(TRUE))
               ^^^^
NameError: name 'TRUE' is not defined. Did you mean: 'True'?
```

**Review by linglp — APPROVED** [review]
LGTM!
